validator/tasks/scheduler.py

```diff
@@ -14,56 +15,48 @@
 from validator.core.models import Task
 from validator.db.sql.tasks import add_task
 from validator.db.sql.tasks import get_tasks_with_status
-from validator.utils.call_endpoint import process_non_stream_get
+from validator.utils.call_endpoint import call_content_service
 
 
 logger = get_logger(name="task synth")
 
 
-async def _get_a_model() -> AsyncGenerator[str, None]:
-    response = await process_non_stream_get(csts.GET_ALL_MODELS_ENDPOINT, None)
+async def _get_models(keypair: Keypair) -> AsyncGenerator[str, None]:
+    response = await call_content_service(csts.GET_ALL_MODELS_ENDPOINT, keypair)
     if not isinstance(response, list):
-        raise TypeError(
-            "Expected a list of responses from GET_ALL_MODELS_ENDPOINT")
+        raise TypeError("Expected a list of responses from GET_ALL_MODELS_ENDPOINT")
     models: list[dict[str, Any]] = response
-    TEMP_MODEL_FAMILIES_ACCEPTED = ['llama', 'falcon', 'mistral', 'gemma', 'gemini',  'phi']
+    TEMP_MODEL_FAMILIES_ACCEPTED = ["llama", "falcon", "mistral", "gemma", "gemini", "phi"]
     model_ids = [
         model.get(csts.GET_ALL_MODELS_ID, "")
-        for model in models 
-        if any(
-            family in model.get(csts.GET_ALL_MODELS_ID, "").lower() 
-            for family in TEMP_MODEL_FAMILIES_ACCEPTED
-        )
+        for model in models
+        if any(family in model.get(csts.GET_ALL_MODELS_ID, "").lower() for family in TEMP_MODEL_FAMILIES_ACCEPTED)
```

Comment:
renamed the function to _get_models and added a keypair to forward to the new call_content_service. Rest if formatting